packages/xai/src/responses/xai-responses-api.ts

```diff
@@ -0,0 +1,211 @@
+import { z } from 'zod/v4';
+
+export type XaiResponsesInput = Array<XaiResponsesInputItem>;
+
+export type XaiResponsesInputItem =
+  | XaiResponsesSystemMessage
+  | XaiResponsesUserMessage
+  | XaiResponsesAssistantMessage;
```

Comment:
there is a lot more, I assume you start out with something minimal and then will add to it?
https://github.com/gr2m/ai-provider-monitor/blob/d8b417d6c8dc5e1a5ae16252fc9bc5f68390b35f/cache/xai/routes/v1/responses/post.json#L40-L463